openmc/mgxs/library.py

```diff
@@ -524,6 +540,20 @@ def build_library(self):
                     mgxs.legendre_order = self.legendre_order
                     mgxs.histogram_bins = self.histogram_bins
 
+                if self.by_nuclide:
+                    try:
+                        domain_nuclides = domain.get_nuclides()
+                    except AttributeError:
+                        domain_nuclides = None
+                    if self.nuclides:
+                        if domain_nuclides:
+                            mgxs.nuclides = [
+                                                nuclide for nuclide in self.nuclides
+                                                if nuclide in domain_nuclides
+                                            ] + ["total"]
```

Comment:
This differs slightly from the PEP8 recommended [hanging indent](https://peps.python.org/pep-0008/#indentation):
```suggestion
                            mgxs.nuclides = [
                                nuclide for nuclide in self.nuclides
                                if nuclide in domain_nuclides
                            ] + ["total"]
```